tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/scoping/integration/TclMacroCallIntegrationTest.xtend

```diff
@@ -0,0 +1,97 @@
+package org.testeditor.tcl.dsl.scoping.integration
+
+import javax.inject.Inject
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.xtext.xbase.compiler.output.ITreeAppendable
+import org.junit.Test
+import org.mockito.Mock
+import org.testeditor.tcl.dsl.jvmmodel.AbstractTclGeneratorIntegrationTest
+import org.testeditor.tcl.dsl.jvmmodel.TclJvmModelInferrer
+
+import static org.mockito.Matchers.*
+import static org.mockito.Mockito.*
+
+class TclMacroCallIntegrationTest extends AbstractTclGeneratorIntegrationTest {
+
+	@Inject TclJvmModelInferrer jvmModelInferrer // class under test
+	@Mock ITreeAppendable outputStub
+
+	override void setup() {
+		super.setup
+		when(outputStub.trace(any(EObject))).thenReturn(outputStub)
+		when(outputStub.append(any(CharSequence))).thenReturn(outputStub)
+	}
+
+	private def parseAmlModel(String aml) {
```

Comment:
This sound like a method we could also put to good use in the super class, couldn't we?